use_case_examples/hybrid_model/serve_model.py

```diff
@@ -0,0 +1,193 @@
+"""Hybrid Model Deployment Server.
+
+Routes:
+    - Get all names
+    - Get client.zip
+    - Add a key
+    - Compute
+"""
+
+import ast
+import io
+import os
+import time
+import uuid
+from collections import defaultdict
+from functools import lru_cache
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, Union
+
+import uvicorn
+from fastapi import FastAPI, Form, HTTPException, UploadFile
+from fastapi.responses import FileResponse, StreamingResponse
+from loguru import logger
+
+# No relative import here because when not used in the package itself
+from concrete.ml.deployment import FHEModelServer
+
+
+def underscore_str_to_tuple(tup):
+    return ast.literal_eval(tup.replace("p_", "(").replace("_p", ")").replace("_", ", "))
```

Comment:
the same function has been defined in `torch/hybrid_model.py` right ? maybe make  a `common/utils.py` ?

Reply:
No it's the reverse function in `hybrid_model.py` 😉

Reply:
ah ok I see my bad

so if I understand correctly, some files (circuits ?) are saved with names based on their shape, using the first function. Then, we load these files and parse their name in order to load back the shapes using `ast.literal_eval`, right ?

but is this safe enough ? are all shapes unique ? And most of all, wasn't there any better solutions than this ? I'm not sure I followed everything but wasn't it possible to have regular namings and retrieve the shapes differently ?

Reply:
Yes for each module we need to have a variety of circuits compiled for each shape.
I found this was the quickest way to get something working.

Safe against what? Well you can't have multiple folders with the same name so yes no issue I believe.

Didn't think of another one, but I am open to suggestions 🙂 
We could have dumped a file on the same and have some sort of unique id but I found it easier to have this information in the folder's name. We can change this afterwards if we find that this creates too many problems.

Reply:
I see, well yes maybe we'll change that later if we think it's not great , thanks for the explanation !